client/src/Register.js

```diff
@@ -8,16 +8,18 @@ import UnexpectedError from './UnexpectedError';
 import ValidationError from './ValidationError';
 
 import './Register.scss';
+import { useAuthContext } from './AuthContext';
 
 function Register() {
+  const authContext = useAuthContext();
   const history = useHistory();
 
+  const [role, setRole] = useState('donor')
```

Comment:
Currently, there is no role attribute in user. So, I created a separate state variable for tracking it, and set an initial value of "donor"...